docker/readmodel-db/catalog.sql

```diff
@@ -0,0 +1,130 @@
+CREATE SCHEMA IF NOT EXISTS readmodel_catalog;
+
+CREATE TABLE IF NOT EXISTS readmodel_catalog.eservice (
+  id UUID,
+  metadata_version INTEGER NOT NULL,
+  producer_id UUID NOT NULL,
+  name VARCHAR NOT NULL,
+  description VARCHAR NOT NULL,
+  technology VARCHAR NOT NULL,
+  created_at TIMESTAMP WITH TIME ZONE NOT NULL,
+  mode VARCHAR NOT NULL,
+  is_signal_hub_enabled BOOLEAN,
+  is_consumer_delegable BOOLEAN,
+  is_client_access_delegable BOOLEAN,
+  template_id UUID,
+  PRIMARY KEY (id),
+  CONSTRAINT eservice_id_metadata_version_unique UNIQUE (id, metadata_version)
+);
+
+CREATE TABLE IF NOT EXISTS readmodel_catalog.eservice_descriptor (
+  id UUID,
+  eservice_id UUID NOT NULL REFERENCES readmodel_catalog.eservice (id) ON DELETE CASCADE,
+  metadata_version INTEGER NOT NULL,
+  version VARCHAR NOT NULL,
+  description VARCHAR,
+  state VARCHAR NOT NULL,
+  audience VARCHAR ARRAY NOT NULL,
+  voucher_lifespan INTEGER NOT NULL,
+  daily_calls_per_consumer INTEGER NOT NULL,
+  daily_calls_total INTEGER NOT NULL,
+  agreement_approval_policy VARCHAR,
+  created_at TIMESTAMP WITH TIME ZONE NOT NULL,
+  server_urls VARCHAR ARRAY NOT NULL,
+  published_at TIMESTAMP WITH TIME ZONE,
+  suspended_at TIMESTAMP WITH TIME ZONE,
+  deprecated_at TIMESTAMP WITH TIME ZONE,
+  archived_at TIMESTAMP WITH TIME ZONE,
+  PRIMARY KEY (id),
+  FOREIGN KEY (eservice_id, metadata_version) REFERENCES readmodel_catalog.eservice (id, metadata_version) DEFERRABLE INITIALLY DEFERRED
+);
+
+CREATE TABLE IF NOT EXISTS readmodel_catalog.eservice_descriptor_template_version_ref (
+  eservice_template_version_id UUID NOT NULL,
+  eservice_id UUID NOT NULL REFERENCES readmodel_catalog.eservice (id) ON DELETE CASCADE,
+  metadata_version INTEGER NOT NULL,
+  descriptor_id UUID NOT NULL REFERENCES readmodel_catalog.eservice_descriptor (id) ON DELETE CASCADE,
+  contact_name VARCHAR,
+  contact_email VARCHAR,
+  contact_url VARCHAR,
+  terms_and_conditions_url VARCHAR,
+  PRIMARY KEY (eservice_template_version_id, descriptor_id),
+  FOREIGN KEY (eservice_id, metadata_version) REFERENCES readmodel_catalog.eservice (id, metadata_version) DEFERRABLE INITIALLY DEFERRED
+);
+
+CREATE TABLE IF NOT EXISTS readmodel_catalog.eservice_descriptor_rejection_reason (
+  eservice_id UUID NOT NULL REFERENCES readmodel_catalog.eservice (id) ON DELETE CASCADE,
+  metadata_version INTEGER NOT NULL,
+  descriptor_id UUID NOT NULL REFERENCES readmodel_catalog.eservice_descriptor (id) ON DELETE CASCADE,
+  rejection_reason VARCHAR NOT NULL,
+  rejected_at TIMESTAMP WITH TIME ZONE NOT NULL,
+  FOREIGN KEY (eservice_id, metadata_version) REFERENCES readmodel_catalog.eservice (id, metadata_version) DEFERRABLE INITIALLY DEFERRED
+);
+
+CREATE TABLE IF NOT EXISTS readmodel_catalog.eservice_descriptor_interface (
```

Comment:
if interfaces need to be downloaded from the public catalog we have 2 options
- allow the public catalog to read S3 objects in core platform (suggested since files are immutable)
- copy all interfaces into the public catalog domain during the scheduled sync

cc @micdes-pagopa